enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,921 @@
+    - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+    ...
+```
+
+When the new `pinnedImages` field is added or changed the machine config
+operator will need to create or update the corresponding
+`/etc/crio/crio.conf.d/pin-upgrade.conf` file, and then reload it using the
+equivalent of `systemctl reload crio.service`.
+
+#### Tool to create the upgrade bundle
+
+The upgrade bundle will be created by an engineer in a partner's factory using a
+new `oc adm upgrade create bundle` command. This engineer will first determine
+the target version number, for example 4.13.7. Note that doing this will
+probably require access to the upgrade service available in api.openshift.com.
+Finding that upgrade version number is outside of the scope of this enhancement.
+
+The engineer will then need internet access and a Linux machine where she can
+run `oc adm upgrade create bundle`, for example:
+
+```bash
+$ oc adm upgrade create bundle \
+--arch=x86_64 \
+--version=4.13.7 \
+--pull-secret=/my/pull/secret.txt \
+--output=/my/bundle/dir
+```
+
+The `oc adm upgrade bundle` command will find the list of image references that
+make up the release, doing the equivalent of this:
+
+```bash
+$ oc adm release info \
+quay.io/openshift-release-dev/ocp-release:4.13.7-x86_64 -o json | \
+jq '.references.spec.tags[].from.name'
+```
+
+In addition to the release images the tool will also support explicitly adding
+custom images. For example:
+
+```bash
+$ oc adm upgrade create bundle \
+--arch=x86_64 \
+--version=4.13.7 \
+--pull-secret=/my/pull/secret.txt \
+--extra-image=quay.io/my-company/my-workload1 \
+--extra-image=quay.io/my-company/my-workload2 \
+...
+--extra-images-file=/my-copany/my-workloads.txt
+...
+--output=/my/bundle/dir
+```
+
+This is intended for situations where the user wants to use the same upgrade
+mechanism for her own images.
+
+The `--extra-image` option will be used to add a single image, and it can be
+repeated multiple times.
+
+The `--extra-images-file` option will be used to add a collection of images
+specified in a text file, one image per line.
+
+The command will then bring up a temporary image registry server, embedded into
+the tool, listening to a randomly selected local port and using a self signed
+certificate. It will then start to copy the images found in the previous step to
+the embedded registry server, using the equivalent of this for each image:
+
+```bash
+$ skopeo copy \
+--src-auth-file=/my/pull.secret.txt \
+--dst-cert-dir=/my/certs \
+docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:... \
+docker://localhost:12345/openshift-release-dev/ocp-v4.0-art-dev@...
+```
+
+When all the images have been copied to the temporary image registry server it
+will be shutdown.
+
+The result will be a directory containing approximately 180 images and requires
+16 GiB of space. The command will then create a `upgrade-4.13.7-x86_64.tar` tar
+file containing that directory and a `metadata.json` file.
```

Comment:
What about instead using something like `oc mirror --config=imageset-config.yaml file://<some-dir>` (through the `oc-mirror` plugin) to copy the specified release version to files? It will also create a desired number of tar archives containing the release images